presto-orc/src/main/java/io/prestosql/orc/OrcReader.java

```diff
@@ -374,10 +387,15 @@ else if (orcType.getOrcTypeKind() == OrcTypeKind.UNION) {
      * Does the file start with the ORC magic bytes?
      */
     private static boolean isValidHeaderMagic(OrcDataSource source)
-            throws IOException
     {
-        Slice headerMagic = source.readFully(0, MAGIC.length());
-        return MAGIC.equals(headerMagic);
+        try {
+            Slice headerMagic = source.readFully(0, MAGIC.length());
+            return MAGIC.equals(headerMagic);
+        }
+        catch (IOException e) {
+            // assume file is an ORC file
```

Comment:
Why the change? Is seems the caller still handles `IOException`

Reply:
This is only used in one place, and this the behavior that is desired there.  I just inlined this so it is easier to understand.